### PR TITLE
sort imports fully deterministically

### PR DESCRIPTION
**Pull request by faddat**
This PR sort imports in a fully 
deterministic fashion using the `gci` linter. 

I tried to follow convention here --

* first the SDK
* then, in this case, more sdk since we've got 
cosmossdk.io and cosmos/cosmos-sdk
* cometbft
* ibc-go

```yaml
  gci:
    sections:
      - standard # Standard section: captures 
all standard packages.
      - default # Default section: contains all 
imports that could not be matched to another 
section type.
      - blank # blank imports
      - dot # dot imports
      - prefix(cosmossdk.io)
      - prefix(github.com/cosmos/cosmos-sdk)
      - prefix(github.com/cometbft/cometbft)
      - prefix(github.com/cosmos/ibc-go)
```

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -2,54 +2,67 @@ run:
   tests: true
   #   # timeout for analysis, e.g. 30s, 5m, default is 1m
   timeout: 5m
+  allow-parallel-runners: true
 
 linters:
   disable-all: true
   enable:
+    - asciicheck
```
Comment (DimitrisJim) [inline .golangci.yml:10]: do we need the rest of these linters? Ideally, since `gci` does seem like a good addition, it would be good to separate from the rest, they could be decided in a separate PR.

File: .golangci.yml [hunk 2]
```diff
@@ -2,54 +2,67 @@ run:
...
     - revive
     - typecheck
     - unconvert
     - unused
     - misspell
+    - zerologlint
 
 issues:
   exclude-rules:
-    - text: "unused-parameter"
```
Comment (DimitrisJim) [inline .golangci.yml:32]: similarly, I think reviewing removing these would be easier.

File: modules/light-clients/06-solomachine/misbehaviour_handle.go [hunk 3]
```diff
@@ -19,7 +19,7 @@ func (cs ClientState) CheckForMisbehaviour(_ sdk.Context, _ codec.BinaryCodec, _
 	return false
 }
 
-func (cs ClientState) verifyMisbehaviour(ctx sdk.Context, cdc codec.BinaryCodec, clientStore sdk.KVStore, misbehaviour *Misbehaviour) error {
+func (cs ClientState) verifyMisbehaviour(cdc codec.BinaryCodec, clientStore sdk.KVStore, misbehaviour *Misbehaviour) error {
```
Comment (DimitrisJim) [inline modules/light-clients/06-solomachine/misbehaviour_handle.go:22]: guess `unused-parameter` being removed complained about this? Looks like a right idea but again I'd think its better if reviewed separately.

**Review by DimitrisJim — COMMENTED** [review]
I think `gci` is a nice addition to have. I'd like to see the other changes (adding linters, removing exclude rules) separated since the addition of gci touches pretty much every go file making the reviewing process already pretty involved.

File: .golangci.yml (hunk 1 above)
Comment (faddat) [reply .golangci.yml:10]: great idea @DimitrisJim

File: modules/light-clients/06-solomachine/misbehaviour_handle.go (hunk 3 above)
Comment (faddat) [reply modules/light-clients/06-solomachine/misbehaviour_handle.go:22]: totally agree, I'll redo as described :)
